### PR TITLE
Feat/lw 11111 remove shared transactions from local storage

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Checklist

- [X] JIRA - \<[link](https://input-output.atlassian.net/browse/LW-11111)>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

- Update lace to use the new addSignatures method in the observable wallet
- Remove storage of shared wallet transaction from local storage

### CODE REVIEW

**Comment by pczeglik-iohk** [comment]
# Allure Report
[`allure-report-publisher`](https://github.com/andrcuns/allure-report-publisher) generated test report!



**processReports**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/smoke/linux/chrome/6600/10293887850/index.html) for [9371dd75](https://github.com/input-output-hk/lace/pull/1355/commits/9371dd7505526d629a7a5a8ef32d4315cb885144)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 32     | 0      | 0       | 0     | 32    | ✅     |

**Review by rhyslbw — CHANGES_REQUESTED** [review]
1. Please add some context to the commit message to explain https://github.com/input-output-hk/lace/pull/1355/commits/9371dd7505526d629a7a5a8ef32d4315cb885144. It's not clear to me what this is fixing and the title seems like it's destructive.
2. `chore` commit type is reserved for changes in the repo outside of any software source code. https://github.com/input-output-hk/lace/pull/1355/commits/5c5c701212f3d2fa825d27ea1ccbfd892d092f38 should be a 
`fix`

**Review by greatertomi — APPROVED** [review]
I tested this and it works as expected. Amazing work @AngelCastilloB 🚀

**Comment by AngelCastilloB** [comment]
> 1. Please add some context to the commit message to explain [9371dd7](https://github.com/input-output-hk/lace/commit/9371dd7505526d629a7a5a8ef32d4315cb885144). It's not clear to me what this is fixing and the title seems like it's destructive. Was the workflow failure caused by the change here? If so it's not atomic so it should be combined with any fix.
> 2. `chore` commit type is reserved for changes in the repo outside of any software source code. [5c5c701](https://github.com/input-output-hk/lace/commit/5c5c701212f3d2fa825d27ea1ccbfd892d092f38) should be a
>    `fix`

@rhyslbw I tidy up the commits and added a bit more context

File: apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/components/SendTransactionDrawer/Footer.tsx [hunk 1]
```diff
@@ -174,19 +169,10 @@ export const Footer = withAddressBookContext(
...
-                body,
-                hash: id
-              },
-              auxiliaryData,
-              bodyCbor: builtTxData.importedSharedWalletTx.body().toCbor()
-            });
-            builtTxData.importedSharedWalletTx.setWitnessSet(
-              Serialization.TransactionWitnessSet.fromCore(mergeWitnesses(txWithOwnSignature.witness, witness))
+            sharedWalletTx = Serialization.Transaction.fromCbor(
+              await inMemoryWallet.addSignatures({ tx: builtTxData.importedSharedWalletTx.toCbor() })
```
Comment (szymonmaslowski) [inline apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/components/SendTransactionDrawer/Footer.tsx:173]: ❤️

**Review by szymonmaslowski — APPROVED** [review]
GJ!

**Review by rhyslbw — APPROVED** [review]
🚀

**Pull request merged**